draft-ietf-tls-tls13.md

```diff
@@ -1805,6 +1806,18 @@ this specification are:
 
 - key_share (see {{key-share}})
 
+In addition, in its updated ClientHello, the client SHOULD NOT offer
+any pre-shared keys associated with a hash other than that of the selected
+cipher suite. This allows the client to avoid having to compute partial
+hash transcripts for multiple hashes in the second ClientHello.
+A client which receives a cipher suite that was not offered MUST abort the handshake.
+Servers MUST ensure that they negotiate the same cipher suite when
+receiving a conformant updated ClientHello (if the server selects the
+cipher suite first, then this will happen automatically). Upon
```

Comment:
(Of course, an implementation could use whatever selection algorithm it likes but, yeah, in general it's expected that the server has the entire ClientHello available and potentially negotiates most things before generating HRR. The orthogonal negotiation helps a lot though and makes shortcuts possible if an implementation decides to go with a sane selection algorithm.)